serialmfg/serial_resources/process_entry.py

```diff
@@ -136,10 +168,65 @@ def add_boolean(self, dataset_name, value, expected_value):
         Returns:
         - API response for adding boolean data
         """
-        dataset = serial.Datasets.get(dataset_name, "CHECKBOX", self.process_id)
+        dataset = None
+        try:
+            dataset = serial.Datasets.get(dataset_name, "CHECKBOX", self.process_id)
+        except Exception as e:
+            if serial.allow_automatic_dataset_creation:
+                dataset = serial.Datasets.create(dataset_name, "BOOLEAN", self.process_id)
+            else:
+                raise e
         data = {"type": "BOOLEAN", "dataset_id": dataset.dataset_id, "value": value, "expected_value": expected_value}
         return self.client.make_api_request(f"/processes/entries/{self.id}", "PUT", data=data)
     
+    def add_link(self, link_name, child_identifier, parent_identifier, break_prior_links=False):
+        """
+        Creates a link between a parent and a child at this specific
+        process_entry
+
+        Args:
+        - link_name: User facing name for the link
+        - child_identifier: Identifier for the child component instance to be linked
+        - parent_identifier: Identifier for the parent component instance to be linked
+        - break_prior_links?: Boolean for whether to break prior links
+
+        Returns:
+        - New component instance link
+        """
+        if serial.debug:
+            print(f"Adding link: {link_name} with child identifier: {child_identifier}")
+        child_component_instance_params = {
+                "identifier": child_identifier
+                }
+        parent_component_instance_params = {
+                "identifier": parent_identifier
+                }
+        link_params = {
+                "name": link_name
+                }
+        child_component_instance = self.client.make_api_request("/components/instances",
+                                                                "GET",
+                                                                params=child_component_instance_params)[0]
+        parent_component_instance = self.client.make_api_request("/components/instances",
+                                                                "GET",
+                                                                params=parent_component_instance_params)[0]
+
+        link_dataset = self.client.make_api_request("/datasets",
+                                                    "GET",
+                                                    params=link_params)[0]
+        process_entry_id = self.id
+        data = {
+                "parent_component_instance_id": parent_component_instance["id"],
+                "child_component_instance_id": child_component_instance["id"],
+                "dataset_id": link_dataset["dataset"]["id"],
+                "process_entry_id": process_entry_id,
+                "break_prior_links": break_prior_links,
+                }
```

Comment:
Are we assuming that the dataset must exist for links? (ie. no automatic dataset creation?)

Reply:
I think that's sensible - links have a lot of complexity and you need more than the dataset. Let me know if you disagree

Reply:
agree